exercises/t_jubb/ex02_oo_basics/src/main.cpp

```diff
@@ -0,0 +1,270 @@
+
+#include <algorithm>
+#include <cmath>
+#include <iostream>
+#include <string>
+#include <vector>
+
+
+/*
+Shape class
+ */
+class Shape {
```

Comment:
When deconstructing problems with O.O. it's good to think of the separation of an abstract "concept" versus a concrete realisation of that concept. For concept types you would generally define a so-called abstract base class, a class where all methods are pure virtual, that itself contains no behaviour but simply defines the interface that any realisation of that concept must adhere to.

For instance, `Shape` is a concept and not a specific realisation so I would expect `Shape` to be abstract and thus something you can't create. A realisation of this or concrete type as they are known would then be a `Square` that inherits from `Shape` and overrides the virtual methods to define its own implementation for each of the methods.